torchrl/trainers/helpers/envs.py

```diff
@@ -428,6 +429,86 @@ def get_stats_random_rollout(
     return stats
 
 
+def generate_stats_from_observation_norms(
+    cfg: "DictConfig",  # noqa: F821
+    proof_environment: EnvBase = None,
+    key: Union[str, Tuple[str, ...]] = None,
+):
+    """Generate stats (loc and scale) of the uninitialized ObservationNorm transforms of a TransformedEnv using its init_stats method.
+
+    Returns a list of (idx, stats) where idx represents the index in the Compose transform of the
+    uninitialized ObservationNorm.
+
+    If the TransformedEnv transform is of type ObservationNorm, the index will be set to 0
+
+    Args:
+        cfg (DictConfig): a config object with `init_env_steps` field, indicating
+            the total number of frames to be collected to compute the stats.
+        proof_environment (EnvBase instance, optional): if provided, this env will
+            be used ot execute the rollouts. If not, it will be created using
+            the cfg object.
+        key (str, optional): if provided, the stats of this key will be gathered.
+            If not, it is expected that only one key exists in `env.observation_spec`.
+
+    """
+    proof_env_is_none = proof_environment is None
+    if proof_env_is_none:
+        proof_environment = transformed_env_constructor(
+            cfg=cfg, use_env_creator=False
+        )()
+
+    if not hasattr(cfg, "init_env_steps"):
+        raise AttributeError("init_env_steps missing from arguments.")
+
+    if (
+        type(proof_environment.transform) != Compose
+        and type(proof_environment.transform) != ObservationNorm
+    ):
+        raise TypeError(
+            f"Expected a Compose or ObservationNorm Transform for the TransformedEnv. "
+            f"Actual {type(proof_environment.transform)}"
+        )
+
+    if key is None:
+        keys = list(proof_environment.observation_spec.keys())
+        key = keys.pop()
+        if len(keys):
+            raise RuntimeError(
+                f"More than one key exists in the observation_specs: {[key] + keys} were found, "
+                "thus generate_stats_from_observation_norm cannot infer which to compute the stats of."
+            )
+
+    obs_norm_transforms = []
+    if type(proof_environment.transform) == Compose:
+        for idx, transform in enumerate(proof_environment.transform):
+            if (
+                type(transform) == ObservationNorm
+                and transform.loc is None
+                and transform.scale is None
+            ):
+                obs_norm_transforms.append((idx, transform))
+    else:
+        obs_norm_transforms.append((0, proof_environment.transform))
+
+    stats = []
+    for (idx, transform) in obs_norm_transforms:
```

Comment:
Upon reflection we could simply take the state dict of the transforms and load it no?
Wouldn't that be simpler than this?
If loc and scale are buffers it could simplify things a bit